app/javascript/helpers/custom-click-handler.js

```diff
@@ -0,0 +1,26 @@
+/** Function to handle the click events by redirecting the actions to js functions. */
+const customJsAction = ({ action }) => {
+  if (action.name === 'miqTreeActivateNode') {
+    const { nodeTree, nodeKey } = action;
+    if (nodeTree && nodeKey) {
+      window.miqTreeActivateNode(nodeTree, nodeKey);
+    }
+  } else if (action.name === 'miqOnClickSelectRbacTreeNode') {
+    if (action.treeId) {
+      window.miqOnClickSelectRbacTreeNode(action.treeId);
+    }
+  }
```

Comment:
This might present nicer as a switch - thoughts?

```suggestion
  switch (action.name) {
  case 'miqTreeActivateNode':
    const { nodeTree, nodeKey } = action;
    if (nodeTree && nodeKey) {
      window.miqTreeActivateNode(nodeTree, nodeKey);
    }
    break;
  case 'miqOnClickSelectRbacTreeNode':
    if (action.treeId) {
      window.miqOnClickSelectRbacTreeNode(action.treeId);
    }
    break;
  }
```

Reply:
Yes, I have been meaning to write a switch case when I encountered a second condition